src/Framework/PaymentGateways/Routes/RouteSignature.php

```diff
@@ -38,12 +60,43 @@ public function toString()
     }
 
     /**
-     * @since 2.19.0
+     * @unreleased
+     *
+     * @return string
+     */
+    public function toHash()
+    {
+        return wp_hash($this->signature);
+    }
+
+    /**
+     * Create expiration timestamp
+     *
+     * @unreleased
+     *
+     * @return string
+     */
+    public static function createExpirationTimestamp()
+    {
+        return (string)current_datetime()->modify('+1 day')->getTimestamp();
```

Comment:
@jonwaldstein `current_datetime ` function added to WP `5.3` and we are compatible with `5.0`. Can we replace this function?

Reply:
Good catch, @ravinderk. Looks like we'll need to roll our own for this one, or bring in the implementation from WP 5.3.

Reply:
It uses PHP's `DateTimeImmutable` and `DateTimeZone` under the hood, so the only missing piece is [wp_timezone_string()](https://developer.wordpress.org/reference/functions/wp_timezone_string/) which is the WordPress specific code introduced in WP 5.3.

Reply:
Working on adding a shim.

Reply:
ah that's annoying, thanks @kjohnson

Reply:
@jonwaldstein @ravinderk See https://github.com/impress-org/givewp/pull/6298/commits/b82e1a4ac0659380b1469e585619cb6a25baff0f

Reply:
@kjohnson awesome! we should probably move that into a `Framework/Support/Shims` since its not gateway specific - but up to you if you want to do that now or later.

Reply:
@jonwaldstein I'll probably go ahead and do that since I need to adjust the loading order anyway.

Reply:
Looks like InPluginUpsells is also using `current_datetime()` 😅 

> Fatal error: Call to undefined function Give\Promotions\InPluginUpsells\current_datetime() in /home/kylejohnson/Code/impress-org/give/src/Promotions/InPluginUpsells/SaleBanners.php on line 59

Reply:
ah yeah lol, I guess we'll need to add a givewp prefix

Reply:
Here is what I've got:

```php
use Give\Framework\Shims\Shim;

public function createExpirationTimestamp()
{
    Shim::load( 'current_datetime' );
    return (string)current_datetime()->modify('+1 day')->getTimestamp();
}
```

Reply:
That works, I just changed to date_create_immutable to avoid exception doc hell

Reply:
@ravinderk @jonwaldstein Tests pass! Just awaiting @Benunc's QA.

Reply:
@ravinderk just re-qreusted your review in light of the resolved feedback.

Reply:
@kjohnson All good 👍